doc/index.md

```diff
@@ -1,56 +1,59 @@
+To use the Npgsql EF Core provider, add a dependencies for `Npgsql` and `Npgsql.EntityFrameworkCore.PostgreSQL`. You can follow the instructions in the general [EF Core Getting Started docs](https://docs.microsoft.com/en-us/ef/core/get-started/).
 
-Following is an example (new-style) csproj using Npgsql EF Core:
+Below is a `.csproj` file for a console application that uses the Npgsql EF Core provider:
 
 ```xml
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="2.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="2.0.0" />
-  </ItemGroup>
-  <ItemGroup>
-    <DotNetCliToolReference Include="Microsoft.EntityFrameworkCore.Tools.DotNet" Version="2.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="2.1.0" />
+    <PackageReference Include="Npgsql" Version="4.0.0" />
+    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="2.1.0" />
   </ItemGroup>
 </Project>
 ```
 
-## Using the Npgsql provider
+## Defining a `DbContext`
 
-    public class BlogContext : DbContext
-    {
-        // When used with ASP.net core, add these lines to Startup.cs
-        //   var connectionString = Configuration.GetConnectionString("BlogContext");
-        //   services.AddEntityFrameworkNpgsql().AddDbContext<BlogContext>(options => options.UseNpgsql(connectionString));
-        // and add this to appSettings.json
-        // "ConnectionStrings": { "BlogContext": "Server=localhost;Database=blog" }
+```c#
+public class BlogContext : DbContext
+{
+    public BlogContext(DbContextOptions<BlogContext> options) : base(options) { }
         
-        public BlogContext(DbContextOptions<BlogContext> options) : base(options) { }
-        
-        public DbSet<BlogPost> BlogPosts { get; set; }     
-    }
-    
+    public DbSet<BlogPost> BlogPosts { get; set; }     
+}
+```
+
+## Additional configuration for ASP.NET Core applications
+
+From `Startup.cs` in `ConfigureServices(IServiceCollection)`:
+
+```c#
+public IServiceProvider ConfigureServices(IServiceCollection services)
+    => services.AddEntityFrameworkNpgsql()
+               .AddDbContext<BlogContext>(options => options.UseNpgsql(connectionString))
+               .BuildServiceProvider();
+```
 ## Using an Existing Database (Database-First)
 
-The Npgsql EF Core provider also supports reverse-engineering a code model from an existing PostgreSQL database ("database-first"). To do so, execute the following if you're using dotnet cli:
+The Npgsql EF Core provider also supports reverse-engineering a code model from an existing PostgreSQL database ("database-first"). To do so, use the `dotnet` CLI to execute the following:
```

Comment:
I think "dotnet CLI" is a pretty standard way to refer to it, so I'd write "To do so, use dotnet CLI to execute the following".